### ISSUE TITLE
Case conflict between SQL and SS (PHP?)

### ISSUE THREAD

**Issue by worikgh** [silverstripe/silverstripe-framework#7882]
## Affected Version

Show version numbers by pasting the output of `composer info --direct`.
Alternatively, hover over the SilverStripe logo in the CMS to basic version information.
$ ../composer.phar info --direct
phpunit/phpunit            5.7.27 The PHP Unit Testing framework.
silverstripe/recipe-cms    1.0.3  SilverStripe recipe for fully featured page and asset content editing
silverstripe/recipe-plugin 1.1.0  Helper plugin to install SilverStripe recipes

## Description
A field in the $db array:
'Rcontent' => 'HTMLText'

IN the getCMSFields as 
        $fields->addFieldToTab('Root.Main', HTMLEditorField::create('RContent', 'Right Hand Column'));

(Note the case: RContent V. Rcontent)

The effect of this is that when I edit 'RContent' in the CMS the data is not written to 'Rcontent' variable.  (Queue much head scratching....).  

Changing teh $db array so it is:
'Rcontent' => 'HTMLText'

and dev/build gives error:

`[Emergency] Uncaught SilverStripe\ORM\Connect\DatabaseException: Couldn't run query: ALTER TABLE "MembershipInformation" ADD "RContent" mediumtext character set utf8 collate utf8_general_ci Duplicate column name 'RContent'`

It is possible this could be fixed locally with  MySQL settings.  So could be a installation option, or case folding column names.

**Comment by dhensby** [comment]
What OS are you running on and what's your MySQL setting for table casing?

The DB should support different casings of the same field.

**Comment by worikgh** [comment]
Ubuntu
$ cat /etc/issue
Ubuntu Bionic Beaver (development branch)

Good question about MySQL.  AFAICT it is whatever the default is.  Which I thought was case insensitive.

Judging from the error that is the case.

When I found this I did some research as I always thought SQL was case insensitive.  SQL *commands* are, but table/column names may or may not be.

**Comment by dhensby** [comment]
@worikgh correct, generally on linux the table/column names are case sensitive and on windows they are not.

You may have your mysql version set to insensitive, which needs to be changed, I'm afraid.

see https://dev.mysql.com/doc/refman/5.7/en/identifier-case-sensitivity.html - the setting for `lower_case_table_names` should be 0.

Also see https://stackoverflow.com/a/6134059, which basically is a copy-paste answer from the mysql docs.

**Comment by dhensby** [comment]
Please can you check your `lower_case_table_names` setting and let me know what it is?

**Comment by tractorcow** [comment]
> Changing teh $db array so it is:
> 'Rcontent' => 'HTMLText'

I assume you mean

```php
'RContent' => 'HTMLText'
```

**Comment by dhensby** [comment]
> We do have a utility for renaming table names based on case, but not table columns. We could probably build that in.

We could, but probably won't ;)
